plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/wizard/newapp/fromtemplate/ApplicationSourceFromTemplateModel.java

```diff
@@ -187,7 +188,15 @@ public void setParameters(List<IParameter> parameters) {
 		  parameters.forEach(p -> paramsMap.put(p.getName(), p.getValue()));
 		}
 		originalValueMap = paramsMap;
-		firePropertyChange(PROPERTY_PARAMETERS, this.parameters, this.parameters = injectProjectParameters(this.eclipseProject, parameters));
+		/*
+		 * firePropertyChange will send an event if the array are differents so we need to check
+		 * otherwise we will risk the ObservableList to be based on the wrong list as it is assigned
+		 * in the call.
+		 */
+		List<IParameter> newParameters = injectProjectParameters(this.eclipseProject, parameters);
+		if (!Objects.equals(this.parameters, newParameters)) {
+	        firePropertyChange(PROPERTY_PARAMETERS, this.parameters, this.parameters = newParameters);
```

Comment:
PropertyChangeSupport calls oldValue.equals(newValue) which for lists compares elements so that it does not matter if it is another instance passed for old value. Sure, if we know that lists contain same elements, we do not need to send the event, but since exactly that is done by pcs, we are spared from doing the check.